libraries/AP_Logger/AP_Logger_File.cpp

```diff
@@ -259,18 +256,17 @@ uint16_t AP_Logger_File::find_oldest_log()
     // We could count up to find_last_log(), but if people start
     // relying on the min_avail_space_percent feature we could end up
     // doing a *lot* of asprintf()s and stat()s
-    hal.scheduler->expect_delay_ms(3000);
+    EXPECT_DELAY(hal, 3000);
     DIR *d = opendir(_log_directory);
-    hal.scheduler->expect_delay_ms(0);
```

Comment:
This is a behaviour change as the delay object will remain in scope well past the point we were zeroing the delay here.

Reply:
yes, and I'm happy with that